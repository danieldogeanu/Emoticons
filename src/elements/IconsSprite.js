import React from 'react';
import { ReactComponent as Icons } from '../images/icons.svg';

const IconsSprite = () => (
	<div
		className="IconsSprite"
		data-testid="IconsSprite"
		style={{display: 'none'}}>
		<Icons />
	</div>
);

export default IconsSprite;
