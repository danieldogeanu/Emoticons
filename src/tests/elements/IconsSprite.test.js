import React from 'react';
import {render} from '@testing-library/react';
import IconsSprite from '../../elements/IconsSprite';

describe('IconsSprite Element', () => {

	it('renders icons sprite properly', () => {
		const testId = 'IconsSprite';
		const {getByTestId, container} = render(<IconsSprite />);
		const renderedSprite = getByTestId(testId);
		const renderedSVG = container.querySelector('svg');

		expect(renderedSprite).toBeInTheDocument();
		expect(renderedSprite).toHaveClass(testId);
		expect(renderedSprite).toHaveStyle('display: none;');
		expect(renderedSprite).toContainElement(renderedSVG);
		expect(renderedSprite).toMatchSnapshot();
	});

});
