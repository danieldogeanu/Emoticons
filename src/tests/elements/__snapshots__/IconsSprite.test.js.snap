// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconsSprite Element renders icons sprite properly 1`] = `
<div
  class="IconsSprite"
  data-testid="IconsSprite"
  style="display: none;"
>
  <svg>
    icons.svg
  </svg>
</div>
`;
